ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -232,6 +228,126 @@ bool RSkComponentImage::processImageData(const char* path, char* response, int s
+//Apply Shadow for transparent image
+    canvas->drawImageRect(imageData, imageTargetRect, &shadowPaint);
+  } else {
+//Apply Shadow for opaque image
+    if(!SaveLayerDone) {
+      canvas->saveLayer(&shadowBounds,&shadowPaint);
+      SaveLayerDone =true;
+    }
+    // clipping done to avoid drawing on non visible area [Area under opque frame]
+    canvas->clipRect(frameBound,SkClipOp::kDifference);
+    shadowPaint.setColor(shadowColor);
+    canvas->drawIRect(shadowFrame, shadowPaint);
+  }
+  if(SaveLayerDone) {
+    canvas->restore();
+  }
+#ifdef SHOW_SHADOW_BOUND
+  SkPaint paint;
+  paint.setStyle(SkPaint::kStroke_Style);
+  paint.setColor(SK_ColorGREEN);
+  paint.setStrokeWidth(2);
+  shadowBounds.join(frameBound);
+  canvas->drawRect(shadowBounds,paint);
+#endif
+}
+
+inline void RSkComponentImage::setPaintFilters (SkPaint &paintObj,const ImageProps &imageProps,
+                                                      SkRect imageTargetRect,SkRect frameRect ,
+                                                      bool  setFilterForShadow, bool opaqueImage) {
+  
+  //This function applies appropriate filter on paint to draw Shadow or Image. 
+
+   /*  Image Filter will be used on below scenario :
+       -------------------------------------------
+      1. For shadow on Image with transparent pixel
+      2. For Image draw with Resize mide as "repeat"
+      3. For Image Draw with with blur Effect.
+   */
+  if((setFilterForShadow && !opaqueImage)||
+      (! setFilterForShadow &&
+         ((imageProps.resizeMode == ImageResizeMode::Repeat) || (imageProps.blurRadius > 0))
+      )) {
+      sk_sp<SkImageFilter> shadowFilter{nullptr};
+      if(setFilterForShadow && (layer()->shadowImageFilter != nullptr) ) {
+         shadowFilter=layer()->shadowImageFilter;
+      }
+      if(imageProps.resizeMode == ImageResizeMode::Repeat) {
+         shadowFilter = (SkImageFilters::Tile(imageTargetRect,frameRect,shadowFilter ));
+      }
+      if(imageProps.blurRadius > 0) {
+        shadowFilter = SkImageFilters::Blur(imageProps.blurRadius, imageProps.blurRadius,shadowFilter);
+      }
+      paintObj.setImageFilter(std::move(shadowFilter));
+   } else if(setFilterForShadow && (shadowMaskFilter != nullptr)) {
+      paintObj.setMaskFilter(shadowMaskFilter);
+  }
+}
+
+inline bool  RSkComponentImage::needsContentShadow(ImageResizeMode resizeMode,
+                                                            bool isOpaque,
+                                                            SkRect frameRect,
+                                                            SkRect imageTargetRect) {
+/* This Function check on whether Image fills the frame or not to decide need for contentShadow 
+   Whether component has visible shadow or not needs to be taken care by the caller.
+*/
+    if(!isOpaque) {
+       /* In case of transparent image, frame size will not be enough to conclude image covers the frame or not
+           So returning false.
```

Comment:
Change the comment here to reflect teh condition. instead of "So returning false." change it to "So returning true."